hopeaisulu/inkubasia-hr-tutorial
# Add shared/ folder and move Employee interface over shared/ folder

Summary:
* Create shared/ folder in the project root
* Move Employee interface into the shared/ folder
* Delete old interface
* Update configurations in tsconfig.json files in server/ & client/ folders
 * Add "paths" to let know ts compiler  where to look when e.g interface is being imported by "@shared/" reference
* Update package.json files in the server/ & client folder, add dependency "@server/shared" & "@client/shared"
 * Under dependencies specify where is the shared/ folder & give it reference name
* Resourses I looked at:
 * https://stackoverflow.com/questions/63922792/nest-js-import-files-from-outside-project-folder
 * https://stackoverflow.com/questions/76143364/nest-js-and-angular-js-monorepo-trying-to-share-library-folder

Test plan:
* There is no behavioural change in this commit
* Delete /dist & /node_modules in the both server/ and client/ folders
 * Install all the dependencies and build the projects again, there shouldn't be any error
 * Run the NestJS server, it should compile without any errors
* Tests should pass
* Test docker if it is still working, it should work
